### PR TITLE
Adding suse ubuntu support

### PR DESCRIPTION
**Pull request by chiqomar**
Using the master branch as the base, and adding openSUSE Leap 15 and 42 in for testing (along with variables for SLES and RedHat that I'm currently using).
The main changes here are

variables for almost everything that can be different from system to system
loading the default.yml first, so those are multiple values don't need to be loaded
Changes to the library file to support Ubuntu
Changes to the Dockerfile to support OpenSUSE Leap (enables systemd)
Addition of 4 more images for molecule testing (hope you've got the space...)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+.pyre/
+.pytest_cache/
+.pytype/
+.ropeproject
+.scrapy
+.spyderproject
+.spyproject
+.tox/
+.venv
+.vscode/*
```
Comment (richm) [inline .gitignore]: why not just `.vscode/`?

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ language: python
 services:
   - docker
 install:
-  - pip install molecule docker
+  - pip install molecule docker ansible-lint flake8
```
Comment (richm) [inline .travis.yml]: this is related to suse or ubuntu support?

File: defaults/main.yml [hunk 3]
```diff
@@ -4,4 +4,4 @@ timesync_ptp_domains: []
 timesync_dhcp_ntp_servers: false
 timesync_step_threshold: -1.0
 timesync_min_sources: 1
-timesync_ntp_provider: ''
+timesync_ntp_provider: ""
```
Comment (richm) [inline defaults/main.yml]: this is related to suse or ubuntu support?

File: library/timesync_provider.sh [hunk 4]
```diff
@@ -6,38 +6,37 @@ is_service_enabled() {
 
 	read -r prev_runlevel runlevel < <(runlevel)
 
-	systemctl is-enabled $name.service &> /dev/null || \
+	systemctl is-enabled $name.service &>/dev/null ||
```
Comment (richm) [inline library/timesync_provider.sh]: is this related to ubuntu suse support?

File: library/timesync_provider.sh [hunk 5]
```diff
@@ -6,38 +6,37 @@ is_service_enabled() {
...
 		chkconfig --list $name 2>/dev/null | grep -q "$runlevel:on"
 }
 
 is_service_active() {
 	local name=$1
 
-	systemctl is-active $name.service &> /dev/null || \
+	systemctl is-active $name.service &>/dev/null ||
```
Comment (richm) [inline library/timesync_provider.sh]: is this related to ubuntu suse support?

File: templates/ntpd.sysconfig.j2 [hunk 6]
```diff
@@ -1,3 +1,3 @@
 # {{ ansible_managed }}
 
-OPTIONS="-g{{ ' -u ntp:ntp -p /var/run/ntpd.pid' if ansible_distribution in ['RedHat', 'CentOS'] and ansible_distribution_major_version|int < 7 else '' }}"
+OPTIONS="-g{{ ' -u ' + timesync_ntp_user + ':' + timesync_ntp_user + ' -p /var/run/ntpd.pid' if ansible_distribution in ['RedHat', 'CentOS'] and ansible_distribution_major_version|int < 7 else '' }}"
```
Comment (richm) [inline templates/ntpd.sysconfig.j2]: Do we need `timesync_ntp_user + ':' + timesync_ntp_group` here or is it always guaranteed that `timesync_ntp_user == timesync_ntp_group`

File: templates/timemaster.conf.j2 [hunk 7]
```diff
@@ -28,12 +28,12 @@ includefile /etc/ntp.conf
...
 path /usr/sbin/chronyd
 {% if timesync_ntp_provider == 'chrony' and __timesync_chrony_version is version('1.30', '<') %}
-options -u chrony
+options -u {{ timesync_chrony_user }}
 {% endif %}
 
 [ntpd]
 path /usr/sbin/ntpd
-options -u ntp:ntp -g
+options -u {{ timesync_ntp_user }}:{{ timesync_ntp_user }} -g
```
Comment (richm) [inline templates/timemaster.conf.j2]: same question here about `timesync_ntp_group`

File: .travis.yml (hunk 2 above)
Comment (richm) [reply .travis.yml]: Also - we are sticking with molecule v2 for now - so please change this to `molecule<3` - and revert the other changes below related to molecule v3 support

File: molecule/default/molecule.yml [hunk 8]
```diff
@@ -3,10 +3,10 @@ dependency:
...
 driver:
   name: docker
-lint:
-  name: yamllint
-  options:
-    config-file: molecule/default/yamllint.yml
+lint: |
+  yamllint -c molecule/default/yamllint.yml .
+  ansible-lint
+  flake8
```
Comment (richm) [inline molecule/default/molecule.yml]: see above - we are sticking with molecule v2 for now

File: molecule/default/molecule.yml [hunk 9]
```diff
@@ -32,7 +41,10 @@ scenario:
...
     - destroy
+provisioner:
+  name: ansible
+  log: true
+  playbooks:
+    converge: ../../tests/tests_default.yml
 verifier:
   name: testinfra
-  lint:
-    name: flake8
```
Comment (richm) [inline molecule/default/molecule.yml]: see above - we are sticking with molecule v2 for now

File: molecule/default/yamllint.yml [hunk 10]
```diff
@@ -9,3 +9,6 @@ rules:
     level: error
   truthy: disable
   document-start: disable
+  line-length:
+    level: warning
+    allow-non-breakable-inline-mappings: true
```
Comment (richm) [inline molecule/default/yamllint.yml]: is this related to ubuntu suse support?
Is this related to molecule v3 support?

File: tasks/main.yml [hunk 11]
```diff
@@ -1,13 +1,15 @@
 ---
+- name: Set default variables
+  include_vars: default.yml
 
 - name: Set version specific variables
-  include_vars: "{{ item }}"
+  include_vars: "{{ item | replace(' ','-') }}"
```
Comment (pcahyna) [inline tasks/main.yml]: interesting - why is this needed, does some distribution have spaces in ansible_distribution or ansible_distribution_version or ansible_distribution_major_version or ansible_os_family ?

File: tasks/main.yml [hunk 12]
```diff
@@ -1,13 +1,15 @@
 ---
+- name: Set default variables
+  include_vars: default.yml
```
Comment (pcahyna) [inline tasks/main.yml]: this should not be needed, if you need to set a variable for all the distributions, set it in vars/main.yml. default.yml is intended for unrecognized distributions.

**Comment by pcahyna** [comment]
still too many unrelated changes, but I pointed this out in the original PR #62 already.

File: tasks/main.yml (hunk 11 above)
Comment (chiqomar) [reply tasks/main.yml]: Yes, see (vars/openSUSE-Leap-15.yml)[https://github.com/chiqomar/timesync/blob/adding-suse-ubuntu-support/vars/openSUSE-Leap-15.yml]

File: templates/ntpd.sysconfig.j2 (hunk 6 above)
Comment (chiqomar) [reply templates/ntpd.sysconfig.j2]: I believe this is the basic configuration, that doesn't typically make sense to change; but, adding a var as default.

**Comment by chiqomar** [comment]
@richm @pcahyna if you all could review this again, I think it's ready for review.

File: .gitignore [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+pip-log.txt
+pip-wheel-metadata/
+profile_default/
+sdist/
+share/python-wheels/
+target/
+var/
+venv.bak/
+venv/
+wheels/
```
Comment (richm) [inline .gitignore]: This is a very impressive list . . .

**Comment by richm** [comment]
[citest]

File: vars/CentOS-6.yml [hunk 14]
```diff
@@ -1 +1,6 @@
+---
 timesync_ntp_provider_os_default: "ntp"
+
+timesync_required_packages:
+  - rpm-python
+  - libselinux-python
```
Comment (richm) [inline vars/CentOS-6.yml]: I don't think any of these changes are required - please revert this file

File: vars/RedHat-6.yml [hunk 15]
```diff
@@ -1 +1,6 @@
+---
 timesync_ntp_provider_os_default: "ntp"
+
+timesync_required_packages:
+  - rpm-python
+  - libselinux-python
```
Comment (richm) [inline vars/RedHat-6.yml]: I don't think any of these changes are required - please revert this file

File: vars/RedHat-7.yml [hunk 16]
```diff
@@ -0,0 +1,4 @@
+---
+timesync_required_packages:
+  - rpm-python
+  - libselinux-python
```
Comment (richm) [inline vars/RedHat-7.yml]: I don't think any of these changes are required - please revert this file

File: vars/RedHat.yml [hunk 17]
```diff
@@ -0,0 +1,4 @@
+---
+timesync_required_packages:
+  - rpm-python
+  - libselinux-python
```
Comment (richm) [inline vars/RedHat.yml]: I don't think any of these changes are required - please revert this file

Comment (richm) [reply vars/RedHat.yml]: In addition - this is the source of the rhel8 test failures, because these packages do not exist on rhel8

File: .gitignore (hunk 13 above)
Comment (chiqomar) [reply .gitignore]: I'm using a VSCode extension that lets you add/append gitignores from here: https://github.com/github/gitignore, specifically ones for Python, Ansible, and MacOS.
It can be verbose, so I did a `sort|uniq` and removed comments.

**Comment by richm** [comment]
ping - any updates?

**Comment by richm** [comment]
@chiqomar do you still want use to merge this?  If so, you'll need to rebase and fix the conflicts.

**Comment by richm** [comment]
[citest pending]

**Comment by richm** [comment]
[citest bad]

**Comment by richm** [comment]
[citest pending]
